docs/DeveloperGuide.md

```diff
@@ -144,16 +144,48 @@ The `Storage` component,
 * inherits from both `AddressBookStorage` and `UserPrefStorage`, which means it can be treated as either one (if only the functionality of only one is needed).
 * depends on some classes in the `Model` component (because the `Storage` component's job is to save/retrieve objects that belong to the `Model`)
 
-### Common classes
+### 3.6 Common classes
 
 Classes used by multiple components are in the `seedu.addressbook.commons` package.
 
 --------------------------------------------------------------------------------------------------------------------
 
-## **Implementation**
+## **4. Implementation**
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### 4.1 Find persons feature
+
+#### 4.1.1 Implementation
+
+This feature allows the user to display selected persons in the contact list. It is facilitated by `ModelManager` which 
+makes use of the method `#updateFilteredPersonList()` to find persons by name or tag.
+
+Given below is an example usage scenario of how the find person mechanism behaves at each step.
+
+Step 1: The user inputs `find n/Alex t/friends` to find selected persons.
+
+Step 2: This argument is passed into `LogicManager` which calls on `Coach2K22Parser#parseCommand()` to find a suitable parser class to process the user inputs. This initialises the `FindPersonCommandParser` where its method `#parse()` is called to process the user inputs. 
+
+Step 3: It then returns a newly initialised `FindPersonCommand` back to the `LogicManager` for command execution.
+
+Step 4: During the command execution, the `ModelManager#updateFilteredPersonList()` is called which updates the GUI display with only selected persons shown in the contact list. The command results are then generated and shown to the user.
+
+The steps above are summarised using a sequence diagram as shown below.
+![FindPersonSequenceDiagram](images/FindPersonSequenceDiagram.png)
```

Comment:
I think there might be a missing : for the CommandResult instance in the diagram!

Reply:
Okay thanks! I have edited the sequence diagram